app/src/main/java/org/p2p/wallet/auth/interactor/restore/SocialShareRestoreInteractor.kt

```diff
@@ -1,14 +1,18 @@
 package org.p2p.wallet.auth.interactor.restore
 
 import org.p2p.wallet.auth.repository.RestoreFlowDataLocalRepository
+import org.p2p.wallet.auth.repository.UserSignUpDetailsStorage
 
 class SocialShareRestoreInteractor(
-    private val restoreFlowDataLocalRepository: RestoreFlowDataLocalRepository
+    private val restoreFlowDataLocalRepository: RestoreFlowDataLocalRepository,
+    private val userSignUpDetailsStorage: UserSignUpDetailsStorage,
 ) {
     fun restoreSocialShare(socialShare: String, socialShareUserId: String) {
         restoreFlowDataLocalRepository.also {
             it.socialShare = socialShare
             it.socialShareUserId = socialShareUserId
+            it.deviceShare =
```

Comment:
why SocialShareInteractor is responsible for restoring a deviceShare too? Can you extract id?